logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/message/GetNotificationsUseCaseTest.kt

```diff
@@ -14,303 +15,321 @@ import com.wire.kalium.logic.data.message.MessageRepository
 import com.wire.kalium.logic.data.notification.LocalNotificationCommentType
 import com.wire.kalium.logic.data.notification.LocalNotificationMessage
 import com.wire.kalium.logic.data.notification.LocalNotificationMessageAuthor
-import com.wire.kalium.logic.data.publicuser.model.OtherUser
+import com.wire.kalium.logic.data.user.SelfUser
 import com.wire.kalium.logic.data.user.UserAvailabilityStatus
 import com.wire.kalium.logic.data.user.UserRepository
 import com.wire.kalium.logic.framework.TestUser
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.util.TimeParserImpl
 import io.mockative.Mock
 import io.mockative.any
-import io.mockative.anything
 import io.mockative.classOf
 import io.mockative.given
 import io.mockative.mock
 import io.mockative.once
 import io.mockative.verify
 import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.TestResult
 import kotlinx.coroutines.test.runTest
-import kotlin.test.BeforeTest
 import kotlin.test.Test
 import kotlin.test.assertEquals
 import kotlin.test.assertTrue
 
 @OptIn(ExperimentalCoroutinesApi::class)
 class GetNotificationsUseCaseTest {
-    @Mock
-    val messageRepository: MessageRepository = mock(classOf<MessageRepository>())
-
-    @Mock
-    val userRepository: UserRepository = mock(classOf<UserRepository>())
-
-    @Mock
-    val conversationRepository: ConversationRepository = mock(classOf<ConversationRepository>())
-
-    private lateinit var getNotificationsUseCase: GetNotificationsUseCase
-
-    @BeforeTest
-    fun setup() {
-        getNotificationsUseCase = GetNotificationsUseCaseImpl(messageRepository, userRepository, conversationRepository)
-    }
 
     @Test
     fun givenEmptyConversationList_thenEmptyNotificationList() = runTest {
-        given(userRepository)
-            .suspendFunction(userRepository::getSelfUser)
-            .whenInvoked()
-            .thenReturn(flowOf(TestUser.SELF))
-        given(conversationRepository)
-            .coroutine { getConversationsForNotifications() }
-            .then { flowOf(listOf()) }
-
-        val notificationsListFlow = getNotificationsUseCase()
+        val (_, getNotifications) = Arrangement()
+            .withSelfUserId()
+            .withSelfUser()
```

Comment:
Ideally it would be better to pass the expected value per parameter to these arrangement config methods to increase visibility. That way it is clearer what is set up and expected instead of "trusting" the arrangement to do the correct thing "under the hood". Something like this:
```suggestion
            .withSelfUser(TestUser.SELF)
```
It does the same thing in both cases, but on this one, the developer knows right away the type of user that will be returned by the api w/o having to look into the method